packages/esm-hiv-app/src/views/hiv-testing-services/tab-list/pre-test.component.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { EncounterListColumn, getObsFromEncounter, EncounterList } from '@ohri/openmrs-esm-ohri-commons-lib';
+import {
+  PreTestApproachConceptUUID,
+  PreTestConceptUUID,
+  PreTestEligibleForTestingConceptUUID,
+  PreTestEncounterTypeConceptUUID,
+  PreTestHIVTestDoneConceptUUID,
+  PreTestPopulationConceptUUID,
+} from '../../../constants';
+import { moduleName } from '../../../index';
+
+interface HIVPreTestTabListProps {
+  patientUuid: string;
+}
+
+const HIVPreTestTabList: React.FC<HIVPreTestTabListProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+
+  const columns: EncounterListColumn[] = useMemo(
+    () => [
+      {
+        key: 'vlDate',
+        header: t('PreTestEncDate', 'Encounter Date'),
+        getValue: (encounter) => {
+          return getObsFromEncounter(encounter, PreTestHIVTestDoneConceptUUID, true);
+        },
+      },
+      {
+        key: 'dsdstatus',
+        header: t('PreTestSetting', 'Setting'),
+        getValue: (encounter) => {
+          return getObsFromEncounter(encounter, PreTestConceptUUID);
+        },
+      },
+      {
+        key: 'dsdModel',
+        header: t('PreTestApproach', 'Approach'),
+        getValue: (encounter) => {
+          return getObsFromEncounter(encounter, PreTestApproachConceptUUID);
+        },
+      },
+      {
+        key: 'dsdModel',
+        header: t('PreTestPltn', 'Population Type'),
+        getValue: (encounter) => {
+          return getObsFromEncounter(encounter, PreTestPopulationConceptUUID);
+        },
+      },
+      {
+        key: 'dsdModel',
+        header: t('PreTestEFT', 'Eliglble for Testing'),
```

Comment:
review all these and make the necessary changes

Reply:
@pirupius Done